### ISSUE TITLE
Log4J Dependency Error with spring boot 2.0 and kafka client 0.11.x

### ISSUE THREAD

**Issue by kganger** [spring-cloud/spring-cloud-stream#1195]
There seems to be a fundamental incompatibility with spring-cloud-streams 2.0.x and kafka clients 0.11.0.x.  Per the documentation these two versions should be compatible.  However, when attempting to start an application in this configuration you get the following error 

Caused by: java.lang.NoClassDefFoundError: org/apache/log4j/Logger
	at kafka.utils.Logging$class.logger

This is the build.gradle file where I downgraded spring-kafka and kafka-clients 
`buildscript {
	ext {
		springBootVersion = '2.0.0.M7'
	}
	repositories {
		mavenCentral()
		maven { url "https://repo.spring.io/snapshot" }
		maven { url "https://repo.spring.io/milestone" }
	}
	dependencies {
		classpath("org.springframework.boot:spring-boot-gradle-plugin:${springBootVersion}")
	}
}

apply plugin: 'java'
apply plugin: 'eclipse'
apply plugin: 'org.springframework.boot'
apply plugin: 'io.spring.dependency-management'

group = 'com.example'
version = '0.0.1-SNAPSHOT'
sourceCompatibility = 1.8

repositories {
	mavenCentral()
	maven { url "https://repo.spring.io/snapshot" }
	maven { url "https://repo.spring.io/milestone" }
}

dependencyManagement {
	imports {
		mavenBom 'org.springframework.cloud:spring-cloud-stream-dependencies:Elmhurst.M3'
	}
}

dependencies {
	compile('org.springframework.boot:spring-boot-starter')
	testCompile('org.springframework.boot:spring-boot-starter-test')
	compile 'org.springframework.cloud:spring-cloud-stream'
	compile 'org.springframework.cloud:spring-cloud-starter-stream-kafka'
	compile group: 'org.apache.kafka', name: 'kafka-clients', version: '0.11.0.2'
	compile group: 'org.springframework.kafka', name: 'spring-kafka', version: '2.0.2.RELEASE'
}
`

**Comment by olegz** [comment]
Unfortunately Kafka still depends on Log4j 1 and spring-boot removed its support for it. We have to manually add it to kafka binder, so please update your binder as well or if you prefer to use the old binder then add the dependency on log4j 1 manually in you gradle script

**Comment by garyrussell** [comment]
The log4j dependency is no longer needed with the latest 2.0.0.BUILD-SNAPSHOT.

The dependency comes from the scala jars, which are no longer needed since the AdminClient in 1.0.0 now supports resizing partitions.

__If you use a version later than 2.0.0.M3 with an 0.11 client, automatic partition resizing won't work__

@olegz I am not sure why the log4j dep is needed any more, except for tests because of the `KafkaEmedded` from `spring-kafka-test` which needs the scala jars.

I excluded it and this works for me...

	<parent>
		<groupId>org.springframework.boot</groupId>
		<artifactId>spring-boot-starter-parent</artifactId>
		<version>2.0.0.M7</version>
		<relativePath/> 
	</parent>

	<properties>
		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
		<project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
		<java.version>1.8</java.version>
		<spring-cloud.version>Finchley.M5</spring-cloud.version>
	</properties>

	<dependencies>
		<dependency>
			<groupId>org.springframework.cloud</groupId>
			<artifactId>spring-cloud-stream</artifactId>
			<version>2.0.0.BUILD-SNAPSHOT</version>
		</dependency>

		<dependency>
			<groupId>org.springframework.cloud</groupId>
			<artifactId>spring-cloud-stream-binder-kafka</artifactId>
			<version>2.0.0.BUILD-SNAPSHOT</version>
		</dependency>

		<dependency>
			<groupId>org.springframework.cloud</groupId>
			<artifactId>spring-cloud-stream-binder-kafka-core</artifactId>
			<version>2.0.0.BUILD-SNAPSHOT</version>
			<exclusions>
				<exclusion>
					<groupId>log4j</groupId>
					<artifactId>log4j</artifactId>
				</exclusion>
			</exclusions>
		</dependency>

		<dependency>
			<groupId>org.apache.kafka</groupId>
			<artifactId>kafka-clients</artifactId>
			<version>0.11.0.2</version>
		</dependency>

		<dependency>
			<groupId>org.springframework.kafka</groupId>
			<artifactId>spring-kafka</artifactId>
			<version>2.0.2.RELEASE</version>
		</dependency>

		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-test</artifactId>
			<scope>test</scope>
		</dependency>
	</dependencies>

**Comment by olegz** [comment]
@garyrussell indeed initially we thought it was just required for test, but then there was this https://github.com/spring-cloud/spring-cloud-stream-binder-kafka/issues/279

**Comment by garyrussell** [comment]
Something's not right then; 2.0.0.BS has no longer any dependency on the Kafka scala jars and thus no dependency on log4j. I'll look into it tomorrow.

**Comment by olegz** [comment]
Closing it as it appears to be duplicate to https://github.com/spring-cloud/spring-cloud-stream-binder-kafka/issues/279 which has already been addressed
